src/main/java/org/alfresco/rest/api/impl/NodeDefinitionMapperImpl.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.rest.api.impl;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.alfresco.error.AlfrescoRuntimeException;
+import org.alfresco.model.ContentModel;
+import org.alfresco.rest.api.NodeDefinitionMapper;
+import org.alfresco.rest.api.model.NodeDefinitionConstraint;
+import org.alfresco.rest.api.model.NodeDefinition;
+import org.alfresco.rest.api.model.NodeDefinitionProperty;
+import org.alfresco.service.cmr.dictionary.ConstraintDefinition;
+import org.alfresco.service.cmr.dictionary.PropertyDefinition;
+import org.alfresco.service.cmr.dictionary.TypeDefinition;
+import org.alfresco.service.cmr.i18n.MessageLookup;
+import org.alfresco.service.namespace.NamespaceService;
+import org.alfresco.service.namespace.QName;
+/**
+ * Maps representations from TypeDefinition to NodeDefinition
+ *
+ * @author gfertuso
+ */
+public class NodeDefinitionMapperImpl implements NodeDefinitionMapper 
+{
+
+    private final List<String> EXCLUDED_NS = Arrays.asList(NamespaceService.SYSTEM_MODEL_1_0_URI);
+    private static final List<QName> EXCLUDED_PROPS = Arrays.asList(ContentModel.PROP_CONTENT);
+
+    @Override
+    public NodeDefinition fromTypeDefinition(TypeDefinition typeDefinition,
+            MessageLookup messageLookup) 
+    {
+        
+        if (typeDefinition == null)
+        {
+            throw new AlfrescoRuntimeException("Undefined definition for the node");
```

Comment:
Add some context to log message - eg. type definition name (and ideally nodeId if you can pass it through) ?

Reply:
This will never happen for node definition functionality, but  because the method is public I added this check. Initially I had a noderef as parameter so I had that information returned in the exception. I don't need anymore the noderef here and if the TypeDefiniton is null I don't have any information to return. Can you suggest a better message, or should I delete the check? I would prefer not to add a parameter only to use it in the exception message.